WIP: document our guidelines/coding conventions (#2616)
scikit-image/scikit-image: GUIDELINES.rst.txt

Guidelines and Coding Conventions
=================================

Import
------

The following import conventions are used throughout the NumPy source
and documentation::

   import numpy as np
   import matplotlib as mpl
   import matplotlib.pyplot as plt

Do not abbreviate ``scipy``. There is no motivating use case to
abbreviate it in the real world, so we avoid it in the documentation
to avoid confusion.

In our examples, we import the ``data`` submodule instead of its function,
such that the image can be changed easily::

    from skimage import data

    img = data.coins()

Arguments and keyword arguments
-------------------------------

* `image`: input image as numpy ndarray
* `selem`: The neighborhood expressed as an array of 1's and 0's
* `multichannel` (default: ): Whether the last axis of the image is to be
interpreted as multiple channels
* `in_place` (default: )
* `max_iter`: Maximum number of iterations
* `tol`: Tolerance
* `method`
* `mode`

Spelling
--------

> We try to use as much as possible the American English spelling:

Reviewer: ```suggestion
For consistency, we choose to use American English spelling:
```

Guidelines and Coding Conventions
=================================

Import
------

The following import conventions are used throughout the NumPy source
and documentation::

   import numpy as np
   import matplotlib as mpl
   import matplotlib.pyplot as plt

Do not abbreviate ``scipy``. There is no motivating use case to
abbreviate it in the real world, so we avoid it in the documentation
to avoid confusion.

In our examples, we import the ``data`` submodule instead of its function,
such that the image can be changed easily::

    from skimage import data

>     img = data.coins()

Reviewer 2: This has to be `image` :)

>     img = data.coins()

Author: @soupault Please, read my comment: https://github.com/scikit-image/scikit-image/issues/2538#issuecomment-295104651 and add your point of view :)

>     img = data.coins()

Reviewer 2: Works fine to me either way. :)
